pkg/registry/core/service/storage/rest.go

```diff
@@ -846,9 +859,77 @@ func (rs *REST) releaseServiceClusterIPs(service *api.Service) (released map[api
 	return rs.releaseClusterIPs(toRelease)
 }
 
+// tests if two preferred dual-stack service have matching ClusterIPFields
+// assumption: old service is a valid, default service (e.g., loaded from store)
+func isMatchingPreferDualStackClusterIPFields(oldService, service *api.Service) bool {
+	if oldService == nil {
+		return false
+	}
+
+	if service.Spec.IPFamilyPolicy == nil {
+		return false
+	}
+
+	// if type mutated then it is an update
+	// that needs to run through the entire process.
+	if oldService.Spec.Type != service.Spec.Type {
+		return false
+	}
+	// both must be type that gets an IP assigned
+	if service.Spec.Type != api.ServiceTypeClusterIP &&
+		service.Spec.Type != api.ServiceTypeNodePort &&
+		service.Spec.Type != api.ServiceTypeLoadBalancer {
+		return false
+	}
+
+	if oldService.Spec.Type != api.ServiceTypeClusterIP &&
+		oldService.Spec.Type != api.ServiceTypeNodePort &&
+		oldService.Spec.Type != api.ServiceTypeLoadBalancer {
+		return false
+	}
```

Comment:
this is a nit, but one of this is redundant since we check that `oldService.Spec.Type == service.Spec.Type`

Reply:
the first if statement 
```golang
 875   if oldService.Spec.Type != service.Spec.Type {                                                                                  
 876     return false                                                                                                                   
 877   }
```
validates that type has changed, if it is then it should go through the process of alloc/release as needed. the following if statements check if old/new is in fact something that we need we need to alloc/release for.

Reply:
@khenidak I think what Antonio is saying is, if we already know that service.Spec.Type == oldService.Spec.Type (because line 875) and then we just ensured that service.Spec.Type is OK, why do we need to check that oldService.Spec.Type is OK? we just checked that (by equivalence to service.Spec.Type) in line 879, right?